compiler/jsgen.nim

```diff
@@ -1634,16 +1634,8 @@ proc genRepr(p: PProc, n: PNode, r: var TCompRes) =
   of tyEnum, tyOrdinal:
     gen(p, n.sons[1], r)
     useMagic(p, "cstrToNimstr")
-    var offset = ""
-    if t.kind == tyEnum:
```

Comment:
You are referring to https://github.com/nim-lang/Nim/commit/a76b74ed74c98eff41752fc9cb71676f527bab21 right?  
You are right, this breaks when the enum offset is negative.  
Wrapping the key in quotes (`\"$1\"` ) to make it a jstring should be enough to fix this. JS object key are internally converted to strings anyway.